### PR TITLE
CON-4038-public extract charity

### PR DESCRIPTION
**Pull request by VasanthaKasirajan3008**

### CODE REVIEW

File: src/SFA.DAS.ReferenceData.Organisations.Database/StoredProcedures/ImportDataFromLoadTables.sql [hunk 1]
```diff
@@ -1,40 +1,28 @@
...
+	IF OBJECT_ID('[CharityData].charity_aoo', 'U') IS NOT NULL DROP TABLE [CharityData].charity_aoo
+	IF OBJECT_ID('[CharityData].class', 'U') IS NOT NULL DROP TABLE [CharityData].class
+	IF OBJECT_ID('[CharityData].class_ref', 'U') IS NOT NULL DROP TABLE [CharityData].class_ref
+	IF OBJECT_ID('[CharityData].financial', 'U') IS NOT NULL DROP TABLE [CharityData].financial
+	IF OBJECT_ID('[CharityData].name', 'U') IS NOT NULL DROP TABLE [CharityData].name
+	IF OBJECT_ID('[CharityData].[objects]', 'U') IS NOT NULL DROP TABLE [CharityData].[objects]
+	IF OBJECT_ID('[CharityData].overseas_expend', 'U') IS NOT NULL DROP TABLE [CharityData].overseas_expend
+	IF OBJECT_ID('[CharityData].partb', 'U') IS NOT NULL DROP TABLE [CharityData].partb
+	IF OBJECT_ID('[CharityData].remove_ref', 'U') IS NOT NULL DROP TABLE [CharityData].remove_ref
+	IF OBJECT_ID('[CharityData].trustee', 'U') IS NOT NULL DROP TABLE [CharityData].trustee
```
Comment (cofaulco) [inline src/SFA.DAS.ReferenceData.Organisations.Database/StoredProcedures/ImportDataFromLoadTables.sql:16]: These should not be needed if we have removed the tables from the dacpac. They will be removed on the next deployment anyway

File: src/SFA.DAS.ReferenceData.Organisations.Database/Scripts/PreDeployment/PreDeployment.sql [hunk 2]
```diff
@@ -8,4 +8,11 @@
...
-*/
\ No newline at end of file
+*/
+ALTER TABLE [CharityImport].[extract_charity]  ALTER COLUMN name VARCHAR (500) NULL;
+ALTER TABLE [CharityImport].[extract_charity]  ALTER COLUMN orgtype VARCHAR (50) NULL;
+
+ALTER TABLE [CharityData].[charity]  ALTER COLUMN name VARCHAR (500) NULL;
+ALTER TABLE [CharityData].[charity]  ALTER COLUMN orgtype VARCHAR (50) NULL;
+
+ALTER TABLE [CharityData].[charitynamesearch]  ALTER COLUMN Name VARCHAR (500) NULL;
```
Comment (cofaulco) [inline src/SFA.DAS.ReferenceData.Organisations.Database/Scripts/PreDeployment/PreDeployment.sql:18]: These are unnecessary, they can simply be changed in the corresponding table .sql files in the DACPAC

File: src/SFA.DAS.ReferenceData.Infrastructure/packages.config [hunk 3]
```diff
@@ -1,8 +1,9 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
 <packages>
   <package id="CsvHelper" version="2.16.3.0" targetFramework="net452" />
-  <package id="Dapper" version="1.50.2" targetFramework="net452" />
+  <package id="Dapper" version="2.0.35" targetFramework="net462" />
   <package id="DFE.Edubase2.SoapAPI.Client" version="0.0.9" targetFramework="net452" />
+  <package id="FastMember" version="1.5.0" targetFramework="net462" />
```
[the remark this answers is not in the document]
Comment (VasanthaKasirajan3008) [reply src/SFA.DAS.ReferenceData.Infrastructure/packages.config:6]: FastMember  used here  ObjectReader.Create(charityImports)

File: src/SFA.DAS.ReferenceData.CharityImport.WebJob/Updater/CharityImporter.cs [hunk 4]
```diff
@@ -37,77 +39,54 @@ public async Task RunUpdate()
             var lastImport = await _charityRepository.GetLastCharityDataImport();
             
             // start at May 2017 which will import June 2017 when no previous import exists
-            var importInfo = await SearchForDownload(lastImport != null 
+            var importInfo = await DownloadPublicCharityFile(lastImport != null
```
Comment (cofaulco) [inline src/SFA.DAS.ReferenceData.CharityImport.WebJob/Updater/CharityImporter.cs:42]: can we change to not physically download the zip to the filesystem and instead return a stream, like here: https://github.com/SkillsFundingAgency/das-courses-api/blob/580b4f1ef42d23a5bdc6f33725815efaee3b4d97/src/SFA.DAS.Courses.Application/CoursesImport/Services/LarsImportStaging.cs#L44

File: src/SFA.DAS.ReferenceData.Organisations.Database/Scripts/PreDeployment/PreDeployment.sql (hunk 2 above)
Comment (cofaulco) [reply src/SFA.DAS.ReferenceData.Organisations.Database/Scripts/PreDeployment/PreDeployment.sql:18]: these don't seem to have been removed.

**Pull request merged**